jetty-core/jetty-compression/jetty-compression-common/src/main/java/org/eclipse/jetty/compression/Compressions.java

```diff
@@ -0,0 +1,64 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.compression;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+import org.eclipse.jetty.util.TypeUtil;
+import org.eclipse.jetty.util.thread.AutoLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Compression} implementations available via {@link java.util.ServiceLoader}.
+ */
+public class Compressions
```

Comment:
I don't think this class is necessary.

In the proxy case, we do not want both client and server to reference the same JVM singleton, but rather have each its own instance, for example to support different configurations.

The client will look the compressions in `HttpClient.doStart()`, and the server could do the same in `CompressionHandler.doStart()`.

Reply:
Removed.